django/db/models/fields/related.py

```diff
@@ -1318,7 +1346,7 @@ def get_db_prep_lookup(self, lookup_type, value, connection, prepared=False):
 
     def is_hidden(self):
         "Should the related object be hidden?"
-        return self.related_name and self.related_name[-1] == '+'
+        return bool(self.related_name and self.related_name[-1] == '+')
```

Comment:
Seems like you only removed the cast in the `is_hidden()` method below in this file.